BeagleBone_black/src/sensors/can_proxi.hpp

```diff
@@ -0,0 +1,79 @@
+/*
+ * Author: Martin Kristien
+ * Organisation: HYPED
+ * Date: 08/06/18
+ * Description: Implements proximity driver over CANBUS using
+ * data from SAMC21 board
+ *
+ *    Copyright 2018 HYPED
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#ifndef BEAGLEBONE_BLACK_SENSORS_CAN_PROXI_HPP_
+#define BEAGLEBONE_BLACK_SENSORS_CAN_PROXI_HPP_
+
+#include <cstdint>
+
+#include "data/data.hpp"
+#include "sensors/interface.hpp"
+#include "utils/io/can.hpp"
+
+
+namespace hyped {
+
+// namespace utils { namespace io { class Can; }}
+namespace utils { class Logger; }
+
+using utils::io::can::Frame;
+using utils::io::CanProccesor;
+using utils::Logger;
+
+namespace sensors {
+
+constexpr uint32_t kNumProximities = data::Sensors::kNumProximities;
+
+
+/**
+ * @brief Represents and instance of a proximity sensor connected to SAMC21 board.
+ * The values are transfered over CANBUS.
+ *
+ * Note, all 8 values are transfered as a single CAN message with a single ID.
+ * Therefore, only 1 object handles CAN message processing and puts the data to
```

Comment:
Does it matter which object handles it? Can the handling be made `static`?

Reply:
The handling is static. However, CAN library needs an actual object registered to find the owner of any can message.
What could work is having an object registered in CAN, and upon calling `object->processNewData()` the CanProxi object would call a static method for processing. I have no preference here, so let me know if you want it in the static way.

Reply:
Oh, I see. The handling functions are only called by `Can` class so the user of `CanProxi` doesn't care at all about the comms; they only call `getData()` and `isOnline()` and the outputs are being updated automatically in the background. I think it's good as it is.